zyte_spider_templates/spiders/ecommerce.py

```diff
@@ -92,7 +93,16 @@ class EcommerceSpider(Args[EcommerceSpiderParams], BaseSpider):
     @classmethod
     def from_crawler(cls, crawler: Crawler, *args, **kwargs) -> scrapy.Spider:
         spider = super(EcommerceSpider, cls).from_crawler(crawler, *args, **kwargs)
-        spider.allowed_domains = [get_domain(spider.args.url)]
+        seed_url = spider.args.seed_url
+        if seed_url:
+            response = requests.get(seed_url)
```

Comment:
Correct me if I'm wrong but would the `crawler` instance here already have access to a good-to-go Downloader? If so, can we use it instead of using the external `requests` library?

Reply:
The goal is for the request to be handled synchronously. Doing otherwise would complicate the code a bit.